DataCapturing/Synchronization/Synchronizer.swift

```diff
@@ -122,12 +126,46 @@ public class Synchronizer {
     */
     public func syncChecked() {
         self.isReachableCheckingQueue.sync {
-            if isReachable(reachabilityManager!.networkReachabilityStatus) {
+            if syncOnWiFiOnly && isReachableOnEthernetOrWifi {
+                sync()
+            } else if isReachable {
```

Comment:
Changed this a little to respect the `syncOnWiFiOnly` attribute.

Reply:
Is this really correct?

case:
syncOnWiFiOnly = true
isReachableOnEthernetOrWifi = false
isReachable (via mobile network) = true

// From the semantics I assume isReachable this is also true when reachable only via mobile network, so we would sync (false positive) in the case described above?
else if (isReachable) { sync }

Reply:
true